examples/go-V4Auth-example.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+    "fmt"
+    "time"
+    "bytes"
+    "encoding/json"
+    "net/http"
+    "io/ioutil"
+    "github.com/aws/aws-sdk-go/aws/credentials"
+    "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+func main() {
+    bucketName := "test-bucket"
+    d := time.Date(2017, 1, 19, 14, 45, 0, 0, time.UTC)
+    startTime := d.UnixNano() / int64(time.Millisecond)
+    type BucketMetricRequest struct {
+        Buckets []string `json:"buckets"`
+        TimeRange [1]int64 `json:"timeRange"`
+    }
+    bucketMetricRequest := BucketMetricRequest{
+        Buckets:   []string{bucketName},
+        TimeRange: [1]int64{startTime},
+    }
+    buf := bytes.NewBuffer([]byte{})
+    enc := json.NewEncoder(buf)
+    enc.Encode(&bucketMetricRequest)
+    request, err := http.NewRequest("POST",
+        fmt.Sprintf("%s/buckets?Action=ListMetrics", "http://localhost:8100"),
+        buf)
+    if err != nil {
+        panic(err)
+    }
+    reader := bytes.NewReader(buf.Bytes())
+    credentials := credentials.NewStaticCredentials("EO4FRH6BA2L7FCK0EKVT",
+        "B5QSoChhLVwKzZG1w2fXO0FE2tMg4imAIiV47YWX", "")
+    signer := v4.NewSigner(credentials)
+    signer.Sign(request, reader, "s3", "us-east-1", time.Now())
+    client := &http.Client{}
+    resp, err := client.Do(request)
+    if err != nil {
+        panic(err)
+    }
+    defer resp.Body.Close()
```

Comment:
Indeed, it works just the same if added at the end of the main function. I chose to use `defer` here because that's the pattern used in the docs. (https://golang.org/pkg/net/http/). I could update that though if it's clearer.